File: components/lattice/tests/test_lattice_element.cc
#define BOOST_TEST_MAIN
#include <boost/test/unit_test.hpp>
#include "components/lattice/lattice_element.h"

const std::string name("foo");
const std::string type("bar");
const std::string attr("baz");
const std::string attr2("baz2");
const double dblval(3.1415);
const std::string strval("qux");
const std::string strval2("qux");
const double tolerance = 1.0e-12;

BOOST_AUTO_TEST_CASE(construct)
{
    Lattice_element lattice_element(type, name);
}

BOOST_AUTO_TEST_CASE(construct2)
{
    Set_default_attributes_fn_map map(get_standard_default_attributes_fn_map());
    Lattice_element lattice_element(type, name, map);
}

BOOST_AUTO_TEST_CASE(get_type)
{
    Lattice_element lattice_element(type, name);
    BOOST_CHECK_EQUAL(type, lattice_element.get_type());
}

BOOST_AUTO_TEST_CASE(get_name)
{
    Lattice_element lattice_element(type, name);
    BOOST_CHECK_EQUAL(name, lattice_element.get_name());
}

BOOST_AUTO_TEST_CASE(has_double_attribute)
{
    Lattice_element lattice_element(type, name);
    BOOST_CHECK(lattice_element.has_double_attribute(attr) == false);
}

BOOST_AUTO_TEST_CASE(set_get_double_attribute)
{
    Lattice_element lattice_element(type, name);
    lattice_element.set_double_attribute(attr, dblval);
    BOOST_CHECK(lattice_element.has_double_attribute(attr));
    BOOST_CHECK_CLOSE(lattice_element.get_double_attribute(attr), dblval, tolerance);
}

BOOST_AUTO_TEST_CASE(get_double_attributes)
{
    Lattice_element lattice_element(type, name);
    lattice_element.set_double_attribute(attr, dblval);
    BOOST_CHECK(lattice_element.get_double_attributes().count(attr) == 1);
}

BOOST_AUTO_TEST_CASE(has_string_attribute)
{
    Lattice_element lattice_element(type, name);
    BOOST_CHECK(lattice_element.has_string_attribute(attr) == false);
}

BOOST_AUTO_TEST_CASE(set_get_string_attribute)
{
    Lattice_element lattice_element(type, name);
    lattice_element.set_string_attribute(attr, strval);
    BOOST_CHECK(lattice_element.has_string_attribute(attr));
    BOOST_CHECK(lattice_element.get_string_attribute(attr) == strval);
}

BOOST_AUTO_TEST_CASE(get_string_attributes)
{
    Lattice_element lattice_element(type, name);
    lattice_element.set_string_attribute(attr, strval);
    BOOST_CHECK(lattice_element.get_string_attributes().count(attr) == 1);
}


BOOST_AUTO_TEST_CASE(copy)
{
    Lattice_element lattice_element(type, name);
    lattice_element.set_double_attribute("one", 1.0);
    lattice_element.set_double_attribute("two", 2.0);
    lattice_element.set_double_attribute("three", 3.0);
    lattice_element.set_string_attribute("foo", "foo");
    lattice_element.set_string_attribute("bar", "bar");
    lattice_element.add_ancestor("ma");
    lattice_element.add_ancestor("grandma");

    Lattice_element copied_lattice_element(lattice_element);

    BOOST_CHECK_EQUAL(copied_lattice_element.get_type(), type);
    BOOST_CHECK_EQUAL(copied_lattice_element.get_name(), name);
    BOOST_CHECK(std::equal(lattice_element.get_double_attributes().begin(),
                    lattice_element.get_double_attributes().end(),
                    copied_lattice_element.get_double_attributes().begin()));
    BOOST_CHECK(std::equal(lattice_element.get_string_attributes().begin(),
                    lattice_element.get_string_attributes().end(),
                    copied_lattice_element.get_string_attributes().begin()));
    BOOST_CHECK(std::equal(lattice_element.get_ancestors().begin(),
                    lattice_element.get_ancestors().end(),
                    copied_lattice_element.get_ancestors().begin()));
}

File: src/synergia/lattice/tests/unused/test_lattice_element.py
#!/usr/bin/env python

import sys
sys.path.append('../../..')
import local_paths

from synergia.lattice import Lattice_element, Lattice
from nose.tools import *

name = "foo"
type = "bar"
attr = "baz"
attr2 = "qux"
dblval = 2.71828
dblval2 = 3.1415
strval = "quux"
strval2 = "corge"

def test_construct():
    lattice_element = Lattice_element(type, name)

def test_get_type():
    lattice_element = Lattice_element(type, name)
    assert_equal(type, lattice_element.get_type())

def test_get_name():
    lattice_element = Lattice_element(type, name)
    assert_equal(name, lattice_element.get_name())

def test_has_double_attribute():
    lattice_element = Lattice_element(type, name)
    assert_equal(lattice_element.has_double_attribute(attr), False)

def test_get_set_double_attribute():
    lattice_element = Lattice_element(type, name)
    lattice_element.set_double_attribute(attr, dblval)
    assert_equal(lattice_element.has_double_attribute(attr), True)
    assert_almost_equal(lattice_element.get_double_attribute(attr), dblval)

def test_get_double_attributes():
    lattice_element = Lattice_element(type, name)
    lattice_element.set_double_attribute(attr, dblval)
    lattice_element.set_double_attribute(attr2, dblval2)
    double_attributes = lattice_element.get_double_attributes()
    assert_equal(len(double_attributes), 2)
    assert_almost_equal(double_attributes[attr], dblval)
    assert_almost_equal(double_attributes[attr2], dblval2)

def test_has_string_attribute():
    lattice_element = Lattice_element(type, name)
    assert_equal(lattice_element.has_string_attribute(attr), False)

def test_get_set_string_attribute():
    lattice_element = Lattice_element(type, name)
    lattice_element.set_string_attribute(attr, strval)
    assert_equal(lattice_element.has_string_attribute(attr), True)
    assert_equal(lattice_element.get_string_attribute(attr), strval)

def test_get_string_attributes():
    lattice_element = Lattice_element(type, name)
    lattice_element.set_string_attribute(attr, strval)
    lattice_element.set_string_attribute(attr2, strval2)
    string_attributes = lattice_element.get_string_attributes()
    assert_equal(len(string_attributes), 2)
    assert_equal(string_attributes[attr], strval)
    assert_equal(string_attributes[attr2], strval2)

def test_add_ancestor():
    lattice_element = Lattice_element(type, name)

    lattice_element.add_ancestor("pa")
    lattice_element.add_ancestor("grandpa")
    ancestors = lattice_element.get_ancestors()
    assert_equal(ancestors[0], "pa")
    assert_equal(ancestors[1], "grandpa")

def test_has_lattice():
    lattice_element = Lattice_element(type, name)
    has_one = lattice_element.has_lattice()
    assert_equal(has_one, False)

lattice_name = "foo_lattice"
def test_set_get_lattice():
    lattice_element = Lattice_element(type, name)
    lattice = Lattice(lattice_name)
    lattice_element.set_lattice(lattice)
    # This is a less-than-optimal test. The test in
    # test_lattice_element.cc is better.
    assert_equal(lattice_element.get_lattice().get_name(),
        lattice.get_name())

